pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -103,15 +107,29 @@ func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachine
 				return false, errors.Wrapf(err, "failed to create machine scope for machine %q", cm.Name)
 			}
 
-			computeSvc := compute.NewService(m.Scope)
+			vmInterface, err := virtualmachines.NewService(m.Scope).Get(context.Background(), &virtualmachines.Spec{Name: m.Machine.Name})
+			if err != nil && vmInterface == nil {
+				klog.V(2).Infof("Machine %q should join the controlplane: false", m.Machine.Name)
+				return false, nil
+			}
 
-			ok, err := computeSvc.MachineExists(m)
 			if err != nil {
 				return false, errors.Wrapf(err, "failed to verify existence of machine %q", m.Name())
 			}
 
-			klog.V(2).Infof("Machine %q should join the controlplane: %t", scope.Machine.Name, ok)
-			return ok, nil
+			vmExtSpec := &virtualmachineextensions.Spec{
+				Name:   "startup_script",
+				VMName: m.Machine.Name,
+			}
+
+			vmExt, err := virtualmachineextensions.NewService(scope.Scope).Get(context.Background(), vmExtSpec)
```

Comment:
```suggestion
			vmExt, err := virtualMachineExtensions.NewService(scope.Scope).Get(context.Background(), vmExtSpec)
```

Reply:
We should capitalize the subsequent words in a name.

Reply:
this is a package name, which should always be lowercased